specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/applicationGatewayWafDynamicManifest.json

```diff
@@ -0,0 +1,172 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2022-05-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/locations/{location}/applicationGatewayWafDynamicManifest": {
```

Comment:
>applicationGatewayWafDynamicManifest [](http://example.com/codeflow?start=86&length=36)

This is your resource type, and a collection GET, but this API is not modeled as a collection GET. Everytime you create a collection GET it also needs a single GET which could be a singleton, more information here:

-> Incomplete or incomplete CRUD: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

-> Singleton: [https://github.com/Azure/azure-resource-manager-rpc/blob/f1fd954cebb3958a2a2ccb62b949bf60491866e0/v1.0/common-api-contracts.md#singleton-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/f1fd954cebb3958a2a2ccb62b949bf60491866e0/v1.0/common-api-contracts.md#singleton-resources)